### PR TITLE
Add option in beta settings to switch on/off products

### LINKED ISSUE #1601: Products teaser release: add option in beta settings
Add option for users to switch on a beta version of products in the settings.

Screenshot 2019-11-22 at 8 24 09 AM

### PR DESCRIPTION
**Pull request by anitaa1990**
Fixes #1601. This PR adds a new option to the beta section in the Settings page for users to opt in/out of the new products tab.

### Changes:
- Added new products switch to the beta features screen.
- Added new flag in `AppPrefs` to display/hide products tab.
- Removed Feature flags: `PRODUCT_LIST` and `PRODUCT_VARIANTS` since they are both  clubbed into the same release.
- Added a new event `SETTINGS_BETA_FEATURES_PRODUCTS_TOGGLED` which has been added to the events spreadsheet and has been validated/registered.

### Screenshots:
(LEFT: products tab is switched on . RIGHT: products tab is switched off)
 . 

### Testing:
- Verify that products tab is only displayed if the product feature switch in Settings page is switched on.
- Verify that product list is displayed, product search works as expected.
- Verify that clicking on a product redirects to product detail page.
- Notice that variants section is displayed in product detail page, for a variable product.
- Verify that clicking on the variants section redirects to the product variants screen.

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/AppSettingsActivity.kt [hunk 1]
```diff
@@ -68,6 +70,10 @@ class AppSettingsActivity : AppCompatActivity(),
         if (v4StatsOptionChanged) {
             setResult(RESULT_CODE_V4_STATS_OPTIONS_CHANGED)
         }
+
+        if (isProductsFeatureOptionChanged) {
```
Comment (nbradbury) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/AppSettingsActivity.kt]: What happens if both the stats & products features are enabled? Using `setResult` will only inform the main activity that one of them has changed. Perhaps a single `RESULT_CODE_BETA_OPTIONS_CHANGED` result would be better, and the main activity could detect that and then check which beta features need to be enabled or disabled?

Comment (anitaa1990) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/AppSettingsActivity.kt]: Good point! I updated the logic in 5d43847 👍

**Comment by nbradbury** [comment]
@anitaa1990 Changes look good, but I ran across one minor issue: if you visit settings from the product list and then disable the products feature, the product list continues to appear (albeit without a tab).

I'm wondering if we can simplify this by just calling `restart()` in the main activity when any of the beta options changes?

Screenshot_1574766788

**Comment by anitaa1990** [comment]
> I'm wondering if we can simplify this by just calling restart() in the main activity when any of the beta options changes?

That's a neat idea! I thought it might affect performance since we would be calling a lot of api's during restart. But actually we're not 😕Most of the It's a lot cleaner! Thanks! Fixed in 04a8ac5 👍

**Comment by nbradbury** [comment]
Looks good! :shipit:

**Pull request merged**